includes/fields/class-fieldtypes-url.php

```diff
@@ -141,8 +141,9 @@ public function convert_input( &$field, $input ) {
 		}
 
 		// Adds the http protocol if it's missing.
-		$url  = esc_url_raw( is_array( $input ) ? trim( $input[0] ) : $input );
-		$text = trim( is_array( $input ) ? sanitize_text_field( $input[1] ) : '' );
+		$url_raw = esc_url_raw( is_array( $input ) ? trim( $input[0] ) : $input );
+		$url     = is_ssl() ? str_replace( 'http://', 'https://', $url_raw ) : $url_raw;
```

Comment:
Since this is for url values saved into website fields in a listing, I'm not sure that is_ssl has any bearing here. I'm not sure that this is a change we should force.

Reply:
The `is_ssl()` works correctly because I tested it on a `http` and `https` environment respectively.

This fixes the issue where a schema-less url will be appended `http://` on a `https` site and saved to database.

Reply:
The links we are changing here are external links that are added by website users adding or editing listings. I don't think we can blanket switch all of these types of urls to https.

Reply:
We are forcing them to go with `http` though. It's better to err on the side of `https`.

If the URL has another schema, such as `tel:`, `mailto:`, etc., this won't take effect. It only applies to schema-less URLs.

Reply:
esc_url_raw only adds http if it was not already included in the url. This update will always force the url to https, even if you try to switch it to http.

Reply:
That's right, this overwrites all URLs that are saved, even those that are intentionally entered as `http://`. I overlooked this.